transport/src/main/java/io/netty/channel/pool/FixedChannelPool.java

```diff
@@ -441,10 +442,19 @@ public void acquired() {
 
     @Override
     public void close() {
+        closeAsync().awaitUninterruptibly();
```

Comment:
It would probably better to just rethrow the interruptedexception and reset the interrupted bit.   You can wrap it in a RuntimeException.

Reply:
Changed. Let me know if this is what you intended by reset (vs. reset as in, clear the interrupted flag).